Tools/scripts/pathfix.py

```diff
@@ -164,12 +181,47 @@ def fix(filename):
     # Return success
     return 0
 
+
+def parse_shebang(shebangline):
+    """takes shebangline and returns tuple with string of shebang's flags
+    and string of shebang's args
+    parse_shebang(b'#!/usr/bin/python -f sfj') --> (b'f',b'sfj')
+    """
+    end = len(shebangline)
+    start = shebangline.find(b' -', 0, end) + 2  # .find() returns index at space
+    if chr(shebangline[start]).isalpha():
```

Comment:
This code looks fragile. It doesn't support `--check-hash-based-pycs=MODE` option.

IMHO it could be simpler: cut the string at " -" without checking what is next.

The Linux kernel simply uses a space to separate the end of the program name and the start of command line arguments. Maybe " -" is maybe safer than just " ", since I'm not sure how the Windows launcher py.exe handles spaces.